frontend/src/types/settings.ts

```diff
@@ -0,0 +1,11 @@
+export type Settings =
+  | {
+      connectedWallet: WalletType | undefined;
+    }
+  | undefined;
+
+// Values must not be changed for backward compatibility
+export enum WalletType {
+  Metamask = 'metamask',
```

Comment:
Nitpick^10:

```suggestion
  MetaMask = 'metamask',
```

Reply:
Not a nitpick! I checked MetaMask lib and found that they use the camel-case version. I thought at the beginning that they deal with it as one word!